app-connector-client/server/routes/apis.js

```diff
@@ -120,7 +121,30 @@ function get(req, res) {
                 res.status(httpResponse.statusCode).type("json").send(body)
             } else {
                 LOGGER.debug("Received API data: %s", JSON.stringify(body))
-                res.status(httpResponse.statusCode).type("json").send(body)
+                body = JSON.parse(body)
+                if (body.events && body.events.spec) {
+                    refParser.dereference(body.events.spec)
+                        .then(function (schema) {
+                            Object.keys(schema.topics).forEach((topicKey) => {
+                                if (schema.topics[topicKey].publish) {
+                                    schema.topics[topicKey].example = openapiSampler.sample(schema.topics[topicKey].publish.payload)
+                                }
+                                else {
+                                    schema.topics[topicKey].example = openapiSampler.sample(schema.topics[topicKey].subscribe.payload)
+                                }
+                            })
+                            body.events.spec = schema;
+                            res.status(httpResponse.statusCode).type("json").send(body)

```

Comment:
what if the response is already send? as this gets executed concurrently

Reply:
why would it be sent without any res.send called?

Reply:
I agree, it will work, still we need to rework it later in any case as we should use promises more proper and do not have so much logic in a single method.
However, let's go with it for now